blocks/procedures.js

```diff
@@ -840,6 +846,21 @@ Blockly.Blocks['procedures_callnoreturn'] = {
         this.dispose(true, false);
         Blockly.Events.setGroup(false);
       }
+    } else if (event.type == Blockly.Events.CHANGE && event.element == 'disabled') {
+      var name = this.getProcedureCall();
+      var def = Blockly.Procedures.getDefinition(name, this.workspace);
+      if (def && def.id == event.blockId) {
+        // in most cases the old group should be ''
```

Comment:
Elaborate on this assumption.  What are the side effects if not true, and maybe hint at the alternative solution of injecting to the undo stack before the other group.